### PR TITLE
[WIP] Adding typeahead

### PR DESCRIPTION
**Pull request by theresaanna**
This pull request adds the Twitter Typeahead library to the site. The typeahead will serve either English or Spanish state names, depending on which site the user is on.

### CODE REVIEW

File: config/gulp/build.js [hunk 1]
```diff
@@ -4,18 +4,29 @@ var del = require('del');
...
 var spawn = require('cross-spawn');
+var rename = require('gulp-rename');
 
 gulp.task('clean-all', function () {
   return del([
     './static/assets/**/*',
   ]);
 });
 
+gulp.task('move-state-files', function () {
```
Comment (theresaanna) [inline config/gulp/build.js]: This is a little ugly. Is there a nicer way for me to do this?

Comment (rogeruiz) [inline config/gulp/build.js]: @theresaanna I think so, Maybe we can use Hugo to generate the JSON for you and use Bloodhound to cache it locally for newer browsers?

Essentially, [using something similar to this](http://gohugo.io/extras/datadrivencontent/) that would compile a `go template` into a `json` file.

**Comment by rogeruiz** [comment]
@theresaanna  Also, noticing that this has changes from #86 which means that this will need to be rebased when that PR is ready to be reviewed **and / or** lands in `staging`. I can help out with any conflicts. ✋

**Comment by maya** [comment]
Can we change the initial input `value` to `placeholder` for: "Select your state or territory"?

**Pull request closed without merging**